### ISSUE TITLE
Incorrect or unclear example of gmvs_to_poes in calc.py and the advanced users manual

### ISSUE THREAD

**Issue by Jorik0** [gem/oq-engine#7070]
In openquake/commonlib/calc.py, an example is given on how to compute mean hazard curves using the _gmvs_to_poes_ function. A similar example is also given in Section 12.1 of the OpenQuake for Advanced Users Documentation.

As far as I can tell, both of these examples do not correctly compute the mean probability of exceedance when there are multiple realizations. To do so,  _gmvs_to_poes_ would have to be called separately for each realization (and a weighted average would have to be computed of all results). In comparison, in the examples  _gmvs_to_poes_ is only called once for each site.

I've tested this myself, and running the example code provided in openquake/commonlib/calc.py indeed gave results that were significantly higher than they should be.

**Comment by mmpagani** [comment]
Thanks for the note. Providing an example could actually help us to better understand your issue. Did you note the comment in the header of the example where it says *"Works in the case of sampling, when the weights are all equal"*?

**Comment by Jorik0** [comment]
I did notice that comment yes. I assumed it meant that the weights of the realizations should be equal (as is the case with the default sampling method)?

As an additional simple test, I modified the EventBasedPSHA demo by setting number_of_logic_tree_samples to 2. I then computed the mean hazard curve with the code provided in the advanced users documentation (I did have to manually copy the gmvs_to_poes function to get this running on my system) and compared the result to the mean hazard curve exported by OpenQuake.

The results differed again (and as expected, the values given by gmvs_to_poes were higher than the values exported by the engine).
